src/torchcodec/share/cmake/TorchCodec/TorchCodecConfig.cmake

```diff
@@ -0,0 +1,185 @@
+      IMPORTED_LOCATION ${_LIB_PATH}
+    )
+    # Removing _LIB_PATH from cache otherwise it won't be updated
+    # on the next call to find_library().
+    unset(_LIB_PATH CACHE)
+
+    list(APPEND TORCHCODEC_VARIANTS "${ffmpeg_major_version}")
+    set(TORCHCODEC_VARIANTS "${TORCHCODEC_VARIANTS}" PARENT_SCOPE)
+endfunction()
+
+if (DEFINED ENV{TORCHCODEC_FFMPEG4_INSTALL_PREFIX} OR
+    DEFINED ENV{TORCHCODEC_FFMPEG5_INSTALL_PREFIX} OR
+    DEFINED ENV{TORCHCODEC_FFMPEG6_INSTALL_PREFIX} OR
+    DEFINED ENV{TORCHCODEC_FFMPEG7_INSTALL_PREFIX})
+    if (UNIX AND NOT APPLE)
+        set(f4_library_file_names
+          libavutil.so.56
+          libavcodec.so.58
+          libavformat.so.58
+          libavdevice.so.58
+          libavfilter.so.7
+          libswscale.so.5
+          libswresample.so.3
+        )
+        set(f5_library_file_names
+          libavutil.so.57
+          libavcodec.so.59
+          libavformat.so.59
+          libavdevice.so.59
+          libavfilter.so.8
+          libswscale.so.6
+          libswresample.so.4
+        )
+        set(f6_library_file_names
+          libavutil.so.58
+          libavcodec.so.60
+          libavformat.so.60
+          libavdevice.so.60
+          libavfilter.so.9
+          libswscale.so.7
+          libswresample.so.4
+        )
+        set(f7_library_file_names
+          libavutil.so.59
+          libavcodec.so.61
+          libavformat.so.61
+          libavdevice.so.61
+          libavfilter.so.10
+          libswscale.so.8
+          libswresample.so.5
+        )
+    endif()
+
+    add_ffmpeg_target(4 "${f4_library_file_names}")
+    add_ffmpeg_target(5 "${f5_library_file_names}")
+    add_ffmpeg_target(6 "${f6_library_file_names}")
+    add_ffmpeg_target(7 "${f7_library_file_names}")
+
+    add_torchcodec_target(4)
+    add_torchcodec_target(5)
+    add_torchcodec_target(6)
+    add_torchcodec_target(7)
+else()
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(TORCHCODEC_LIBAV IMPORTED_TARGET
+        libavdevice
+        libavfilter
+        libavformat
+        libavcodec
+        libavutil
+        libswresample
+        libswscale
+    )
+
+    if (TARGET PkgConfig::TORCHCODEC_LIBAV)
+        # Split libavcodec's version string by '.' and convert it to a list
+        string(REPLACE "." ";" libavcodec_version_list ${TORCHCODEC_LIBAV_libavcodec_VERSION})
+        # Get the first element of the list, which is the major version
+        list(GET libavcodec_version_list 0 libavcodec_major_version)
+
+        if (${libavcodec_major_version} STREQUAL "58")
+            set(ffmpeg_major_version "4")
+        elseif (${libavcodec_major_version} STREQUAL "59")
+            set(ffmpeg_major_version "5")
+        elseif (${libavcodec_major_version} STREQUAL "60")
+            set(ffmpeg_major_version "6")
+        elseif (${libavcodec_major_version} STREQUAL "61")
+            set(ffmpeg_major_version "7")
+        endif()
```

Comment:
Same here, the logic in this whole branch should ideally be unified with https://github.com/meta-pytorch/torchcodec/blob/75a3325f016483ae1a6af8516b804cc72da39ea6/src/torchcodec/_core/CMakeLists.txt#L276-L315

Reply:
Ok, I've missed this one. I will need to look into it.